lib/benchee/system.ex

```diff
@@ -32,4 +37,62 @@ defmodule Benchee.System do
     end
   end
 
+  @doc """
+  Returns the number of cores available for the currently running VM.
+  """
+  def num_cores do
+    System.schedulers_online()
+  end
+
+  @doc """
+  Returns an atom representing the platform the VM is running on.
+  """
+  def os do
+    {_, name} = :os.type()
+    os(name)
+  end
+  defp os(:darwin), do: :macOS
+  defp os(:nt), do: :Windows
+  defp os(_), do: :Linux
+
+  @doc """
+  Returns a string with detailed information about the CPU the benchmarks are
+  being performed on.
+  """
+  def cpu_speed, do: cpu_speed(os())
+
+  defp cpu_speed(:Windows), do: "N/A"
+  defp cpu_speed(:macOS), do: system_cmd("sysctl", ["-n", "machdep.cpu.brand_string"])
+  defp cpu_speed(:Linux) do
+    "model name\t:" <> cpu_info = to_string(:os.cmd('cat /proc/cpuinfo | grep "model name"'))
+    String.trim(cpu_info)
+  end
+
+  @doc """
+  Returns an integer with the total number of available memory on the machine
+  running the benchmarks.
+  """
+  def available_memory, do: available_memory(os())
+
+  defp available_memory(:Windows), do: "N/A"
+  defp available_memory(:macOS) do
+    {memory, _} = Integer.parse(system_cmd("sysctl", ["-n", "hw.memsize"]))
+    "#{memory / 1_000_000_000} GB"
+  end
+  defp available_memory(:Linux) do
+    "MemTotal:" <> memory = to_string(:os.cmd('cat /proc/meminfo | grep "MemTotal"'))
```

Comment:
huh, why do we use `:os.cmd` instead of `System.cmd`/our system_cmd helper here? Ideally I thought/hoped that we might use that one and its error handling for all the OSs, but probably that was me being too optimistic :sweat_smile: . A generalized interface would still be nice if we can achieve it :)